cli/cmd/install.go

```diff
@@ -361,6 +361,28 @@ func renderCRDs(w io.Writer, options valuespkg.Options, format string) error {
 	}
 	defaultValues["cliVersion"] = k8s.CreatedByAnnotationValue()
 
+	// If any of the Gateway API CRDs are installed, we default to rendering the
+	// Gateway API CRDs.
+	if k != nil {
+		defaultValues["installGatewayAPI"] = false
+		httpRouteCRD, err := k.Apiextensions.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "httproutes.gateway.networking.k8s.io", metav1.GetOptions{})
+		if err == nil && httpRouteCRD != nil && httpRouteCRD.Annotations[k8s.CreatedByAnnotation] != "" {
+			defaultValues["installGatewayAPI"] = true
+		}
+		grpcRouteCRD, err := k.Apiextensions.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "grpcroutes.gateway.networking.k8s.io", metav1.GetOptions{})
+		if err == nil && grpcRouteCRD != nil && grpcRouteCRD.Annotations[k8s.CreatedByAnnotation] != "" {
+			defaultValues["installGatewayAPI"] = true
+		}
+		tlsRouteCRD, err := k.Apiextensions.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "tlsroutes.gateway.networking.k8s.io", metav1.GetOptions{})
+		if err == nil && tlsRouteCRD != nil && tlsRouteCRD.Annotations[k8s.CreatedByAnnotation] != "" {
+			defaultValues["installGatewayAPI"] = true
+		}
+		tcpRouteCRD, err := k.Apiextensions.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "tcproutes.gateway.networking.k8s.io", metav1.GetOptions{})
+		if err == nil && tcpRouteCRD != nil && tcpRouteCRD.Annotations[k8s.CreatedByAnnotation] != "" {
+			defaultValues["installGatewayAPI"] = true
+		}
+	}
```

Comment:
Is this more simply expressed as something like?
```suggestion
        crds := k.Apiextensions.ApiextensionsV1().CustomResourceDefinitions()
        names := []string{
            "httproutes.gateway.networking.k8s.io",
            "grpcroutes.gateway.networking.k8s.io",
            "tlsroutes.gateway.networking.k8s.io",
            "tcproutes.gateway.networking.k8s.io",
        }
        for _, name := range names {
            crd, err := crds.Get(ctx, name, metav1.GetOptions{})
            if err == nil && crd != nil && crd.Annotations[k8s.CreatedByAnnotation] != "" {
                defaultValues["installGatewayAPI"] = true
                break
            }
        }
	}
```

Importantly, we can short-circuit this, right?

Reply:
related, do we really want to silently ignore errors? can't we more narrowly detect NotFound errors?

Reply:
Also, why is it appropriate for us to install Gateway api resources if _something else_ has installed it? don't we care about the value of that annotation?

Reply:
Oh: the "created by" annotation is a linkerd.io namespaced annotation, so it's sufficient. Carry on.